src/ledger/LedgerManagerImpl.h

```diff
@@ -175,6 +175,27 @@ class LedgerManagerImpl : public LedgerManager
         AbstractLedgerTxn& ltx,
         std::unique_ptr<LedgerCloseMetaFrame> const& ledgerCloseMeta);
 
+    UnorderedSet<LedgerKey> getReadWriteKeysForStage(ApplyStage const& stage);
+
+    std::unique_ptr<ThreadEntryMap> collectEntries(AppConnector& app,
+                                                   AbstractLedgerTxn& ltx,
+                                                   Cluster const& cluster);
+
+    std::pair<RestoredKeys, std::unique_ptr<ThreadEntryMap>> applyThread(
+        AppConnector& app, std::unique_ptr<ThreadEntryMap> entryMapByCluster,
+        Cluster const& cluster, Config const& config,
+        SorobanNetworkConfig const& sorobanConfig,
+        ParallelLedgerInfo const& ledgerInfo, Hash const& sorobanBasePrngSeed,
+        SorobanMetrics& sorobanMetrics);
```

Comment:
Some of the `SorobanMetrics` methods aren't thread safe. Those methods should assert that the thread is main or ledger close.

Reply:
Alternatives to this - 
1. Collect metrics within each thread, and update `SorobanMetrics` at the end.
2. Mark the non-thread safe member variables as atomic.

Reply:
It should be fairly simple to make this thread-safe. You could also switch to medida counters, those are thread-safe.